harmony/harmony.py

```diff
@@ -364,8 +376,150 @@ def progress(self, job_id: str) -> int:
         can't be reached.
         """
         session = self._session()
-        response = session.get(self._status_url(job_id)).result()
+        response = session.get(self._status_url(job_id))
         if response.ok:
             return int((response.json())['progress'])
         else:
             response.raise_for_status()
+
+    def wait_for_processing(self, job_id: str, show_progress: bool = False) -> None:
+        """Retrieve a submitted job's completion status in percent.
+
+        Args:
+            job_id: UUID string for the job you wish to interrogate.
+
+        Returns:
+            The job's processing progress as a percentage.
+
+        :raises Exception: This can happen if an invalid job_id is provided or Harmony services
+        can't be reached.
+        """
+        check_interval = 3.0  # in seconds
+        ui_update_interval = 0.33  # in seconds
+        intervals = int(check_interval / ui_update_interval)
+        if show_progress:
+            with progressbar.ProgressBar(max_value=100, widgets=progressbar_widgets) as bar:
+                progress = None
+                check_cycle = cycle([True] + list(repeat(False, intervals - 1)))
+                for should_get in check_cycle:
+                    if should_get:
+                        progress = self.progress(job_id)
+                    bar.update(progress)
```

Comment:
I'm curious what this code is actually meant to do.  `progress` can't change values except when line 406 gets called, right?

Reply:
It's the spinner - see above comment.